### DISCUSSION TITLE
ESP32-CAM and lightsleep()

### DISCUSSION THREAD

**Discussion by davefes** [micropython/micropython#15142] [General]
I have been battling variable picture quality and I don't think it is the camera distro that I am using.
https://github.com/shariltumin/esp32-cam-micropython-2022

It seems that if I use `lightsleep()` that my problems start.

Other places where I use lightsleep, for output pins I hold their state before going to lightsleep and release it when coming out of lightsleep.

Do any of the camera GPIO need the same thing done to them if they are output pins?

Thank you.

**Comment by bidrohini** [comment]
If light sleep is causing issues with the camera, you may need to reinitialize the camera after waking up from light sleep. 

import camera

def init_camera():
    camera.deinit()

    camera.init()

init_camera()


machine.lightsleep(time_in_ms)


init_camera()

To confirm if light sleep is the cause of the problem, try running your code without entering light sleep and see if the picture quality improves. If the issue disappears, then it's likely that light sleep is affecting your camera's operation. For a better connection, you can make a board for the ESP32-Cam like this: https://www.pcbway.com/project/shareproject/ESP32_CAM_Adaptor_PCB.html

**Reply by davefes** [reply]
Thank you for your response.

I suspect that `lightsleep()` is the issue.  My constraint is that I do not want to do a `camera.init()' after waking-up because I have discovered that to get consistent pictures you need to wait for the camera to setup before taking the picture.  Generally, the pictures can be way over-exposed if you don't wait.

I found several comments on other forums about waiting up to 5 seconds after you call `camera.init()`.  By that stage the intruder is long-gone!

Edit: One camera system appears to work with `lightsleep()` and another system does not.  I may be battling quality issues.
